src/App.jsx

```diff
@@ -14,17 +14,24 @@ function App() {
         "Lesson 1.5 Side effects, custom hooks, fragments",
     ]
     const initialBlogs = initialTitles.map(title => titleToBlog(title))
-    const [ blogTitles, setBlogTitles]= useSemiPersistentState('savedBlogTitles', initialBlogs)
+    const [blogTitles, setBlogTitles] = useSemiPersistentState('savedBlogTitles', initialBlogs)
     const [searchText, setSearchText] = react.useState('');
+    const [focus, setFocus] = react.useState('search');
 
+    // focus needs to be reset after the latest focus is applied
+    useEffect(() => {
+        !!focus && setFocus()
+    }, [focus])
 
+    const onAddBlogTitle = (newTitle) => {
+        setBlogTitles([...blogTitles, titleToBlog(newTitle)])
+        setFocus('addBlog')
```

Comment:
We set the focus back into the add blog title input after the form is submitted.